pro/src/hooks/__specs__/useIsOffererValidated.spec.tsx

```diff
@@ -0,0 +1,70 @@
+import { waitFor } from '@testing-library/react'
+import React from 'react'
+
+import { api } from 'apiClient/api'
+import { useIsOffererValidated } from 'hooks/index'
+import { IUseCurrentUserReturn } from 'hooks/useCurrentUser'
+import { individualOfferOffererFactory } from 'utils/individualApiFactories'
+import { renderWithProviders } from 'utils/renderWithProviders'
+
+const TestComponent = (): JSX.Element | null => {
+  const isOffererValidated = useIsOffererValidated()
+
+  if (isOffererValidated) {
+    return <span data-testId="offerer-validated">Validated</span>
+  }
+  return null
+}
+jest.mock('apiClient/api', () => ({
+  api: {
+    listOfferersNames: jest.fn(),
+  },
+}))
+
+jest.mock('hooks/useCurrentUser', () =>
+  jest.fn().mockReturnValue({
+    currentUser: {
+      id: 'EY',
+      isAdmin: false,
+    },
+  } as IUseCurrentUserReturn)
+)
+
+describe('useIsOffererValidated', () => {
+  test('Api calls should be made to check user validation', async () => {
+    renderWithProviders(<TestComponent />)
+    await waitFor(() => {
+      expect(api.listOfferersNames).toHaveBeenCalledTimes(1)
+    })
+    expect(api.listOfferersNames).toHaveBeenCalledWith(undefined, true)
+  })
+
+  describe('offerer not validated', () => {
+    beforeEach(() => {
+      jest.spyOn(api, 'listOfferersNames').mockResolvedValue({
+        offerersNames: [],
+      })
+    })
+
+    test('It should retrun false as the offerer is not validated', async () => {
+      const component = renderWithProviders(<TestComponent />)
+      expect(
+        component.queryByTestId('offerer-validated')
+      ).not.toBeInTheDocument()
+    })
+  })
+
+  describe('offerer validated', () => {
+    beforeEach(() => {
+      const individualOffererNames = individualOfferOffererFactory()
+      jest.spyOn(api, 'listOfferersNames').mockResolvedValue({
+        offerersNames: [individualOffererNames],
+      })
+    })
+
+    test('It should retrun true as the offerer is not validated', async () => {
```

Comment:
typo + is Validated non ?

et du coup dans le expect ça devrait pas être toBeInTheDocument dans ce cas ?

Reply:
Corrigé.